aea/multiplexer.py

```diff
@@ -612,7 +612,11 @@ def setup(
         :return: None.
         """
         self.default_routing = default_routing or {}
+
+        # replace connections
         self._connections = []
+        self._id_to_connection = {}
```

Comment:
How is this fixing it? Can you give any context @solarw ?

Reply:
setup method replaces all connections added before,  `self._id_to_connection` is just mapping for `self._connections`
it's populated with add_connection, but not reset on setup.

i think `setup` method quite tricky, would be nice to get rid out of it.

Reply:
Thanks!